src/library_webgl2.js

```diff
@@ -723,21 +723,41 @@ var LibraryWebGL2 = {
     GL.syncs[id] = null;
   },
 
+#if WASM_BIGINT
+  glClientWaitSync__sig: 'iiii',
+#else
   glClientWaitSync__sig: 'iiiii',
   glClientWaitSync__deps: ['$convertI32PairToI53'],
-  glClientWaitSync: function(sync, flags, timeoutLo, timeoutHi) {
+#endif
+  glClientWaitSync: function(sync, flags, {{{ defineI64Param('timeout') }}}) {
     // WebGL2 vs GLES3 differences: in GLES3, the timeout parameter is a uint64, where 0xFFFFFFFFFFFFFFFFULL means GL_TIMEOUT_IGNORED.
     // In JS, there's no 64-bit value types, so instead timeout is taken to be signed, and GL_TIMEOUT_IGNORED is given value -1.
     // Inherently the value accepted in the timeout is lossy, and can't take in arbitrary u64 bit pattern (but most likely doesn't matter)
     // See https://www.khronos.org/registry/webgl/specs/latest/2.0/#5.15
-    return GLctx.clientWaitSync(GL.syncs[sync], flags, convertI32PairToI53(timeoutLo, timeoutHi));
+    return GLctx.clientWaitSync(GL.syncs[sync], flags,
+#if WASM_BIGINT
+      Number(timeout)
+#else
+      convertI32PairToI53(timeout_low, timeout_high)
+#endif
```

Comment:
This can be replaced with `{{{ receiveI64ParamAsI53('timeout', Infinity) }}}`?

Reply:
(Or, if I'm missing something, we could add a new helper function for this pattern, as it seems like it should be common?)

Reply:
Yes, see the definition of `fd_pwrite` in `src/library_wasi.js` for how we do this elsewhere.

Reply:
Unofrunately `receiveI64ParamAsI53` emits a check for `bigintToI53Checked` or a `convertI32PairToI53Checked`, followed by a IsNaN, which are both undesirable in this case. The desired code is to call `convertI32PairToI53` like show in the PR.

Reply:
Why you explain why?  (i.e. why is convertI32PairToI53Checked undesirable here?)   It seems sad to have these generic mechanism and then not be able to use them.

Reply:
The values in this case are timeouts, so if user specifies a long timeout, we are very ok to round here, and not abort on if rounding occurred. Also code size.

Reply:
According to my math 2^^53 nanoseconds is about 104 days.    If we cap the values here doesn't that just mean that we don't support timeout longer than 104 days?   If that was the only downside it doesn't seem bad, and keeping the code simple might be better?

Or are there some magic values above 2^^53 that we need to support?

Reply:
That is true. I don't know of any magic values, but I would like to stick to this smaller code size form, since in this case expected rounding and smaller code size align with each other (rounding is desired).

Reply:
Ok, added a new function `receiveI64ParamAsI53Unchecked` now, how does that look?